src/fu-engine.c

```diff
@@ -1829,6 +1858,102 @@ fu_engine_get_action_id_for_device (FuEngine *self,
 	return "org.freedesktop.fwupd.update-internal";
 }
 
+/**
+ * fu_engine_get_action_id_for_store:
+ * @self: A #FuEngine
+ * @device_id: A device ID
+ * @store: The #AsStore with the firmware metadata
+ * @flags: The #FwupdInstallFlags, e.g. %FWUPD_DEVICE_FLAG_UPDATABLE
+ * @error: A #GError, or %NULL
+ *
+ * Gets the PolicyKit action ID to use for the install operation.
+ *
+ * Returns: string, e.g. `org.freedesktop.fwupd.update-internal-trusted`
+ **/
+const gchar *
+fu_engine_get_action_id_for_store (FuEngine *self,
+				    const gchar *device_id,
+				    AsStore *store,
+				    FwupdInstallFlags flags,
+				    GError **error)
+{
+	FuDevice *device;
+	g_autofree gchar *guids = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	const gchar *action = NULL, *action_old = NULL;
+	g_autoptr(GString) str = g_string_new (NULL);
+	gboolean matched = FALSE;
+
+	/* Got a specific device */
+	if (g_strcmp0 (device_id, FWUPD_DEVICE_ID_ANY) != 0) {
+		device = fu_device_list_get_by_id (self->device_list, device_id, error);
+		if (device == NULL)
+			return NULL;
+		return fu_engine_get_action_id_for_device (self, device,
+							   store, flags, error);
+	}
+
+	/* Got a wild card */
+	devices = fu_device_list_get_all (self->device_list);
+	for (guint i = 0; i < devices->len; i++) {
+		g_autoptr(GError) error_local = NULL;
+		device = g_ptr_array_index (devices, i);
+		if (fu_engine_store_get_app_by_guids (store, device) != NULL) {
```

Comment:
You could just test for NULL and do `continue` which gets you an extra tab-stop back.

Reply:
fixed